tests/basic/pure_test.go

```diff
@@ -4238,3 +4238,105 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
+		4.Delete the FADA app
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("CreateCloneOfTheFADAVolume", "Create Clone of the FADA Volume and verify the status", nil, 0)
+	})
+	itLog := "Create Clone of the FADA Volume and verify the status and check creation of cloned volume in FA backend"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		var contexts []*scheduler.Context
+		var volumeName string
+		var cloneVolumeId string
+		flashArrays, err := GetFADetailsUsed()
+		log.FailOnError(err, "Failed to get FA details used")
+		stepLog := "Deploy FADA app"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			taskName := "deploy-fada"
+			context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+				AppKeys:            []string{"fio-fa-davol"},
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+				Namespace:          taskName,
+			})
+			log.FailOnError(err, "Failed to schedule application of namespace [%v]", taskName)
+			contexts = append(contexts, context...)
+		})
+		ValidateApplications(contexts)
+
+		stepLog = "Get the corresponding Px volume for the PVC and clone the volume "
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for _, context := range contexts {
+				appsvols, err := Inst().S.GetVolumes(context)
+				log.FailOnError(err, "Failed to get volumes for app %s", context.App.Key)
+				log.InfoD("Starting the Clone of the Volume")
+				for _, vol := range appsvols[:1] {
+					cloneVolumeId, err := Inst().V.CloneVolume(vol.ID)
+					log.FailOnError(err, "Failed to clone volume [%v]", vol.ID)
+					log.InfoD("Clone Volume ID [%v] for parent volume [%v]", cloneVolumeId, vol.ID)
+				}
+			}
+			log.InfoD("Get the corresponding volume name for the volId")
+			for _, context := range contexts {
+				appsvols, err := Inst().S.GetVolumes(context)
+				log.FailOnError(err, "Failed to get volumes for app %s", context.App.Key)
+				for _, vol := range appsvols {
+					if vol.ID == cloneVolumeId {
+						if vol.Name != "" {
+							volumeName = vol.Name
+							log.InfoD("Volume Name for the Clone Volume is [%v]", volumeName)
+						}
+						break
+
+					}
+				}
+			}
+		})
+		checkVolumeExistsInFlashArrays := func(volumeName string, flashArrays []pureutils.FlashArrayEntry) error {
+			cloneVolFound := false
+
+			for _, fa := range flashArrays {
+				faClient, err := pureutils.PureCreateClientAndConnect(fa.MgmtEndPoint, fa.APIToken)
+				log.FailOnError(err, fmt.Sprintf("Failed to connect to FA using Mgmt IP [%v]", fa.MgmtEndPoint))
+				volName, err := GetVolumeCompleteNameOnFA(faClient, volumeName)
+				log.FailOnError(err, fmt.Sprintf("Failed to get volume name for volume [%v]", volumeName))
+				log.Infof("Name of the Volume is [%v]", volName)
+
+				isExists, err := pureutils.IsFAVolumeExists(faClient, volName)
+				log.FailOnError(err, fmt.Sprintf("Failed to check if volume exists on FA: %v", err))
+
+				if isExists {
+					cloneVolFound = true
```

Comment:
setting this variable is not needed as once the volume is Exists function will anyways return Nil